### ISSUE TITLE
Support fstrings in PyPy 3.5

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/python-language-server#956]
@skamansam commented on [Thu Apr 11 2019](https://github.com/Microsoft/vscode-python/issues/5210)

## Environment data
- VSCode Info: Version: 1.34.0-insider
                         Commit: 0ab39f4148f242e7b0802330385fc99b4845aa31
                         Date: 2019-04-08T05:13:56.940Z
                         Electron: 3.1.8
                         Chrome: 66.0.3359.181
                         Node.js: 10.2.0
                         V8: 6.6.346.32
                         OS: Linux x64 5.0.5-200.fc29.x86_64 (Fedora 29)
- VS Code version: 1.34.0-insider (commit 0ab39f414)
- Extension version (available under the Extensions sidebar): 2019.3.6558 (8 April 2019)
- OS and version: Linux x64 5.0.5-200.fc29.x86_64 (Fedora 29)
- Python version (& distribution if applicable, e.g. Anaconda): pypy3.5-6.0.0
- Type of virtual environment used (N/A | venv | virtualenv | conda | ...): pyenv 
- Relevant/affected Python packages and their versions: XXX

## Expected behaviour

When using the new format string specifier, the code should be valid, and not display any problem. 

## Actual behaviour

When using the new format string specifier, the code is not valid, and displays the problems:
```
invalid syntax Python(parser-16) [24, 54]
Undefined variable: 'f' Python(undefined-variable) [24, 53]
```

In this screenshot, you can see the issues:
![screenshot-parser-16-error](https://user-images.githubusercontent.com/102971/55983934-45de5c00-5c6a-11e9-823a-da1274f3bb8d.png)

## Steps to reproduce:

1. create a new python file
2. paste the following into the file: 
```
print(f"I like the number {3}.")
```
3. save and wait a few minutes (it takes a bit for the errors to be displayed.)
## Logs
```
{
	"resource": "[...]/elastic_mixin.py",
	"owner": "_generated_diagnostic_collection_name_#1",
	"code": "parser-16",
	"severity": 8,
	"message": "invalid syntax",
	"source": "Python",
	"startLineNumber": 24,
	"startColumn": 54,
	"endLineNumber": 24,
	"endColumn": 71
}
{
	"resource": "[...]/elastic_mixin.py",
	"owner": "_generated_diagnostic_collection_name_#1",
	"code": "undefined-variable",
	"severity": 4,
	"message": "Undefined variable: 'f'",
	"source": "Python",
	"startLineNumber": 24,
	"startColumn": 53,
	"endLineNumber": 24,
	"endColumn": 54
}
``` 

Output from `Console` under the `Developer Tools` panel (toggle Developer Tools on under `Help`; turn on source maps to make any tracebacks be useful by running `Enable source map support for extension debugging`)

```
( no output )
```

**Comment by jakebailey** [comment]
This isn't a bug; their interpreter is set to Python 3.5. fstrings were introduced in 3.6, and our parser won't accept syntax not present in the specified version of Python.

**Comment by jakebailey** [comment]
It seems that PyPy backported f-strings from 3.6 to its 3.5 distribution, so the version string doesn't actually reflect the language version supported by the interpreter.

**Comment by MikhailArkhipov** [comment]
Not sure how many users specifically use 3.5 vs 3.6

**Comment by jakebailey** [comment]
The PyPy release with this 3.5 backport doesn't appear to be supported upstream. I'm going to close this; I don't think this is something we should fix at this point given everyone's moved on.

**Comment by jmwoloso** [comment]
What is the fix for this? I'm running a 3.7.6 venv in vs code and f-strings throw syntax errors.

**Comment by MikhailArkhipov** [comment]
If you are running 3.7, please open separate issue. This one is about 3.5.
